File: 2018/day1/day1.py
arr = []
output = 0
found = False
results = [0]
checks = 0

def calc(a, b):
    global output
    global found
    global checks
    if a == "+":
        output = output + b
    else:
        output = output - b

    if output in results:
        print(f"found {output}")
        found = True

    results.append(output)


with open("input.txt", "r") as f:
    data = f.readlines()
    for x in data:
        arr.append(x.replace("\n", ""))

while not found:
    for x in arr:
        calc(x[0], int(x[1:]))
    checks += 1
    if checks % 25 == 0:
        print(results)
    print(checks)

print(output)
print(results)

File: 2020/Ruby/Day1/day1.rb
f= File.open("input.txt")
data = f.readlines.map(&:chomp)
f.close

data.each do |x|
  data.each do |y|
    data.each do |z|
      if x.to_i != y.to_i and y.to_i != z.to_i and (x.to_i + y.to_i + z.to_i) == 2020
        puts (x.to_i * y.to_i * z.to_i)
        exit
      end
    end
  end
end


